### PR TITLE
Adding init flad to skip adding repos

### LINKED ISSUE #7868: Skip adding stable/incubator on init flag
In Helm v2 when `helm init` is run it adds the stable repo and an update happens. Even if you specify the tiller image. This is a problem for air gaped environments. They can specify a local tiller image but still have issues due to the upstream charts repo.

Since we are [extending the Helm v2 support window](https://helm.sh/blog/covid-19-extending-helm-v2-bug-fixes/) it may be worth adding a flag to `init` to skip trying to add the stable repo. 

Note, this was captured in the [CNCF Air Gapped Working Group](https://docs.google.com/document/d/1TLCvhy283FMNupQNxkfmQGjE2um2Pnl-piA_lTRsh9w/edit#heading=h.8ngolbigvi7q).

### PR DESCRIPTION
**Pull request by mattfarina**
The stable charts repository is nearing the end of its life. Once
that has happened those who attempt to run helm init will get an
error. This flag provides a means to still use Helm v2 after the
stable charts repository is gone.

A secondary situation is that Helm v2 can be used in environmnets
without a network connection.

Closes #7868

**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains unit tests
- [x] this PR has been tested for backwards compatibility

### CODE REVIEW

File: cmd/helm/init.go [hunk 1]
```diff
@@ -118,6 +119,7 @@ func newInitCmd(out io.Writer) *cobra.Command {
 	f.BoolVarP(&i.clientOnly, "client-only", "c", false, "If set does not install Tiller")
 	f.BoolVar(&i.dryRun, "dry-run", false, "Do not install local or remote")
 	f.BoolVar(&i.skipRefresh, "skip-refresh", false, "Do not refresh (download) the local repository cache")
+	f.BoolVar(&i.skipRepos, "skip-repo", false, "Skip adding the stable and local repositories")
```
Comment (bacongobbler) [inline cmd/helm/init.go:122]: should this be `--skip-repos` to align with the variable name `skipRepos`? Given this skips adding multiple repositories, I think the plurality makes sense here

**Review by bacongobbler — COMMENTED** [review]
looks great! Just had one question about the flag name. LGTM otherwise

File: cmd/helm/init.go (hunk 1 above)
Comment (mattfarina) [reply cmd/helm/init.go]: oops. fixed. good catch.

**Review by bacongobbler — APPROVED** [review]

**Comment by mattfarina** [comment]
This is a feature addition so I moved it to the v2.17.0 release. We have a couple things going in that release including a new tiller location in support of deprecation.

**Comment by dosuken123** [comment]
@mattfarina Thank you for working on this! It seems currently [pre-release version `v2.17.0-rc.1`](https://github.com/helm/helm/releases/tag/v2.17.0-rc.1) is only available. Do you know when the official release is going to happen?

**Pull request merged**